src/Query/GroupDisplayHeadingSelector.ts

```diff
@@ -95,16 +95,16 @@ export class GroupHeadings {
      * Calculate the minimal set of headings that should be displayed
      * before the tasks with the given group names.
```

Comment:
Since we are discussing the general state of this code and what it does, do you understand the comments in lines 95 & 96 here? I honestly struggle to.

Reply:
> Since we are discussing the general state of this code and what it does, do you understand the comments in lines 95 & 96 here? I honestly struggle to.

The best explanation I could come up with is here:

https://github.com/obsidian-tasks-group/obsidian-tasks/blob/main/src/Query/GroupHeadings.ts#L4-L82

(As an aside, from the refactoring in this PR I have some ideas about how to simplify all this code...)